packages/kv-store/src/keyv/keyv.ts

```diff
@@ -156,6 +158,39 @@ export class Keyv<Value = any> extends EventEmitter implements KeyvStore<Value>
       .then(() => Promise.all(results))
   }
 
+  private isMapWithEntries(store: any): store is Map<string, Value> {
+    return store instanceof Map && typeof store.entries === 'function'
+  }
+
+  private async getFromMapWithPrefix(
+    prefixes: Array<string>,
+    options?: {
+      raw?: boolean
+    },
+  ): Promise<Array<KeyvStoredData<Value>>> {
+    if (!this.isMapWithEntries(this.store)) {
+      return []
+    }
+
+    const map = this.store as Map<string, Value>
+    const entries = Array.from(map.entries())
+
+    const result = []
+    for (const prefix of prefixes) {
```

Comment:
can we not clean this stuff up with a reduce?

something like this

const map = this.store as Map<string, Value>;
const entries = Array.from(map.entries());

const result = prefixes.reduce((acc, prefix) => {
  const matchingEntry = entries.find(([key]) => key.startsWith(prefix));
  if (matchingEntry) {
    acc.push(matchingEntry[1] as KeyvStoredData<Value>);
  } else {
    acc.push(undefined);
  }
  return acc;
}, []);

return result;